collectors/plugins.d/plugins_d.c

```diff
@@ -616,6 +620,39 @@ inline size_t pluginsd_process(RRDHOST *host, struct plugind *cd, FILE *fp, int
             enabled = 0;
             break;
         }
+        else if(likely(hash == LABEL_HASH && !strcmp(s, PLUGINSD_KEYWORD_LABEL))) {
+            debug(D_PLUGINSD, "requested a LABEL CHANGE");
+            char *store;
+            if(!words[4])
+                store = words[3];
+            else {
+                store = callocz(PLUGINSD_LINE_MAX + 1, sizeof(char));
+                char *move = store;
+                int i = 3;
+                while (i < w) {
+                    size_t length = strlen(words[i]);
+                    memcpy(move, words[i], length);
+                    move += length;
+                    *move++ = ' ';
+
+                    i++;
+                    if(!words[i])
+                        break;
+                }
+            }
+
+            new_labels = add_label_to_list(new_labels, words[1], store, strtol(words[2], NULL, 10));
+        }
+        else if(likely(hash == OVERWRITE_HASH && !strcmp(s, PLUGINSD_KEYWORD_OVERWRITE))) {
+            debug(D_PLUGINSD, "requested a OVERWITE a variable");
+            if(!host->labels) {
+                host->labels = new_labels;
+            } else {
+                clean_label_list(host, new_labels);
+            }
+
+            new_labels = NULL;
+        }
```

Comment:
I'll tag comments on this line, but really they apply to the whole of this block.

I don't understand what you are doing here, or why - there is no description in the issue of what you are trying to achieve. It looks like you are adding some kind of feature that will overwrite the entire set of host labels and I don't understand which requirement this would come from. I think we should probably discuss whether or not this is a requirement of host labels and their support in streaming.

In addition - it is not a good idea to replace the `host->labels` without holding the write lock. Even if the value is null it could be overwritten by a CLI generated reload operation?

Reply:
By steps, every time that the stream sends a new group of labels, we will update that labels on master, but we are not overwritten the master label, it is the slave database instance that is changed.

We had two options to do this, but in any case the protocol would need to be updated. Case I decided to use the existent keywords I would need to add conditions inside the previous parser to identify a label or a chart value, or I could create two new commands that would be responsible exclusively for labels. I opted for the second option, because I thought it would be better to have different commands for charts and labels, because the data types are different.